### PR TITLE
Remove Student Conferences Model

### LINKED ISSUE #535: Remove student conferences model
we can use `groups` instead of the `student_conferences` model to keep track of a student's conference.

### PR DESCRIPTION
**Pull request by maxwolffe**
Fix #535

### CODE REVIEW

File: app/assets/javascripts/components/student/student_conference.es6.jsx [hunk 1]
```diff
@@ -32,21 +32,17 @@ class StudentConference extends Component {
   // --------------------------------------------------
   // Render
   // --------------------------------------------------
+  // TODO: Add Student status and Rooming information.
   render() {
-    var studentConference = this.props.student.student_conference;
+    var studentConference = this.props.student.group.conference;
```
Comment (warrenshen) [inline app/assets/javascripts/components/student/student_conference.es6.jsx]: Just change this variable name to `conference`?

File: app/assets/javascripts/components/student/student_conference.es6.jsx [hunk 2]
```diff
@@ -32,21 +32,17 @@ class StudentConference extends Component {
...
     return (
       <div style={this.styles.container}>
         <h4>{'Status'}</h4>
-        <h6>{studentConference.status}</h6>
         <br />
         <h4>{'Group'}</h4>
-        <h6>{studentConference.group_letter}</h6>
-        <h6>{studentConference.team_leader_name}</h6>
-        <h6>{studentConference.junior_crew_name}</h6>
+        <h6>{studentConference.name}</h6>
```
Comment (warrenshen) [inline app/assets/javascripts/components/student/student_conference.es6.jsx]: I'm assuming you'll be rendering the other data later through `this.props.student.group`?
Might be better to do `var group = this.props.student.group` then.

File: app/assets/javascripts/stores/student_store.es6.jsx [hunk 3]
```diff
@@ -10,7 +10,7 @@
       this.sidebar = true;
       this.student = {
         school: {},
-        student_conference: {},
+        conference: {},
```
Comment (warrenshen) [inline app/assets/javascripts/stores/student_store.es6.jsx]: Alphabetize, and a student doesn't have a `conference`, it has a `group`!

File: app/models/conference.rb [hunk 4]
```diff
@@ -12,8 +12,6 @@
 
 class Conference < ActiveRecord::Base
 
-  has_many :groups
-  has_many :student_conferences, dependent: :destroy
-  has_many :students, through: :student_conferences
+  has_many :groups, dependent: :destroy
```
Comment (warrenshen) [inline app/models/conference.rb]: Nice.

File: app/models/student.rb [hunk 5]
```diff
@@ -33,8 +33,7 @@ class Student < ActiveRecord::Base
   has_many :comments, dependent: :destroy
   has_many :visits, dependent: :destroy, as: :visitable
 
-  has_one :student_conference
-  has_one :conference, through: :student_conference
+  has_one :conference, through: :group
```
Comment (warrenshen) [inline app/models/student.rb]: Did you try seeing if this works in the rails console with `rails c`?

Comment (maxwolffe) [inline app/models/student.rb]: works!

**Pull request merged**